hash_writer.go

```diff
@@ -33,157 +33,249 @@ type HashWriter interface {
+	if error != nil {
+		return error
+	}
+	return hashWriter.ionWriter.WriteBool(val)
 }
 
 func (hashWriter *hashWriter) WriteInt(val int64) error {
-	hashWriter.ionWriter.WriteInt(val)
-	panic("implement me")
+	error := hashWriter.hashScalar(ion.IntType, val)
+	if error != nil {
+		return error
+	}
+	return hashWriter.ionWriter.WriteInt(val)
 }
 
 func (hashWriter *hashWriter) WriteUint(val uint64) error {
-	hashWriter.ionWriter.WriteUint(val)
-	panic("implement me")
+	error := hashWriter.hashScalar(ion.IntType, val)
+	if error != nil {
+		return error
+	}
+	return hashWriter.ionWriter.WriteUint(val)
 }
 
 func (hashWriter *hashWriter) WriteBigInt(val *big.Int) error {
-	hashWriter.ionWriter.WriteBigInt(val)
-	panic("implement me")
+	error := hashWriter.hashScalar(ion.IntType, val)
+	if error != nil {
+		return error
+	}
+	return hashWriter.ionWriter.WriteBigInt(val)
 }
 
 func (hashWriter *hashWriter) WriteFloat(val float64) error {
-	hashWriter.ionWriter.WriteFloat(val)
-	panic("implement me")
+	error := hashWriter.hashScalar(ion.FloatType, val)
+	if error != nil {
+		return error
+	}
+	return hashWriter.ionWriter.WriteFloat(val)
 }
 
 func (hashWriter *hashWriter) WriteDecimal(val *ion.Decimal) error {
-	hashWriter.ionWriter.WriteDecimal(val)
-	panic("implement me")
+	error := hashWriter.hashScalar(ion.DecimalType, val)
+	if error != nil {
+		return error
+	}
+	return hashWriter.ionWriter.WriteDecimal(val)
 }
 
 func (hashWriter *hashWriter) WriteTimestamp(val time.Time) error {
-	hashWriter.ionWriter.WriteTimestamp(val)
-	panic("implement me")
+	error := hashWriter.hashScalar(ion.TimestampType, val)
+	if error != nil {
+		return error
+	}
+	return hashWriter.ionWriter.WriteTimestamp(val)
 }
 
 func (hashWriter *hashWriter) WriteSymbol(val string) error {
-	hashWriter.ionWriter.WriteSymbol(val)
-	panic("implement me")
+	error := hashWriter.hashScalar(ion.SymbolType, val)
+	if error != nil {
+		return error
+	}
+	return hashWriter.ionWriter.WriteSymbol(val)
 }
 
 func (hashWriter *hashWriter) WriteString(val string) error {
-	hashWriter.ionWriter.WriteString(val)
-	panic("implement me")
+	error := hashWriter.hashScalar(ion.StringType, val)
+	if error != nil {
+		return error
+	}
+	return hashWriter.ionWriter.WriteString(val)
 }
 
 func (hashWriter *hashWriter) WriteClob(val []byte) error {
-	hashWriter.ionWriter.WriteClob(val)
-	panic("implement me")
+	error := hashWriter.hashScalar(ion.ClobType, val)
+	if error != nil {
+		return error
+	}
+	return hashWriter.ionWriter.WriteClob(val)
 }
 
 func (hashWriter *hashWriter) WriteBlob(val []byte) error {
-	hashWriter.ionWriter.WriteBlob(val)
-	panic("implement me")
+	error := hashWriter.hashScalar(ion.BlobType, val)
+	if error != nil {
+		return error
+	}
+	return hashWriter.ionWriter.WriteBlob(val)
 }
 
 func (hashWriter *hashWriter) BeginList() error {
+	error := hashWriter.stepIn(ion.ListType)
+	if error != nil {
+		return error
+	}
+
 	hashWriter.ionWriter.BeginList()
 	panic("implement me")
```

Comment:
Looks like you forgot to remove this panic.

Reply:
Removed